app/javascript/packs/shared/actions/UsersActions.js

```diff
@@ -35,79 +49,216 @@ export function getCurrentUser() {
   };
 }
 
-export function changeFullName(name) {
+export function saveFullName({ fullName }) {
   return {
     type: CHANGE_CURRENT_USER_FULL_NAME,
-    payload: name
+    payload: fullName
   };
 }
 
-export function changeInitials(initials) {
+export function changeFullName(name) {
+  return dispatch => {
+    axios
+      .post(CHANGE_USER_FULL_NAME_PATH, {
+        withCredentials: true,
+        fullName: name
+      })
+      .then(({ data }) => {
+        dispatch(saveFullName(data));
+      })
+      .catch(err => console.log(err));
+  };
+}
+
+export function saveInitials({ initials }) {
   return {
     type: CHANGE_CURRENT_USER_INITIALS,
     payload: initials
   };
 }
 
-export function changeEmail(email) {
+export function changeInitials(initials) {
```

Comment:
Did we decide to put everything in redux?

Reply:
No. But I feel if we will be building multiple react apps it is better to make json response with lettle more data and make this settings app feel more native.

If you think this is wrong I can rewrite this to use state only.

Reply:
so you can do that when the user submit the changes you trigger a post request that also returns the other user data eq: email, ..... be aware you to not send the user password :smile: and with that data you hidrate the state and get the component rerender. Don't worry too much about speed here probably is not such a problem if you send 4 key value pairs instead of one

Reply:
ok so still redux here :smile: If we chose to take this path it will be probably good to discuss how we will handle the data on frontend trough sciNote. Will we have one storage tree/ multiple storages/ multiple trees/... how we solve the relations trough different objects we have a relational data model now and we have to parse it in a "document data model" or will we have like multiple key/value pairs to hold data (objects eg: task, experiment, permissions, ...) and some to hold the app state eg.: modal show/hide, flash show/hide, which addon is enabled .... We could go with multiple storages for every view and have a separate storage for stuff like modals, flash messages,... you can ask Luka for another opinion

Reply:
Good point. We should discuss this open ended questions :+1: